data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -140,6 +144,21 @@ public KafkaCustomConsumer(final KafkaConsumer consumer,
         this.errLogRateLimiter = new LogRateLimiter(2, System.currentTimeMillis());
     }
 
+    public KafkaCustomConsumer(final KafkaConsumer consumer,
```

Comment:
Usually, we do reverse way when multiple constructors are present. Invoke the second constructor from the first by passing `null` as the compression config. 

```
constructor (...existing args..., compression-arg) {
   init all the members
}

constructor(..existing args...) {
    this(existing args, null);
}
```